ir/metadata_types.go

```diff
@@ -0,0 +1,37 @@
+package ir
+
+import "time"
+
+// MetadataType interface implemented by metadata field types
+type metadataType interface {
+	update(with metadataType) metadataType // Determines how the metadata is updated when the node is updated.
```

Comment:
This "legacy code" from the initial design, where we had `AssemblyTime` and `TTL` as metadata types (and I was hoping to make easy for developers to register new metadata in their Node). But as you suggest, I probably overthought it. 

I agree that now this doesn't make sense anymore with a single metadata type, and specially if we consider cross-updates between metadata types. I can remove it if you think it is confusing rather than helping.

Reply:
I think it's fine to keep it for now. This is internal, so it does not confuse our users.